include/inria_wbc/behaviors/humanoid/walk.hpp

```diff
@@ -0,0 +1,66 @@
+#ifndef IWBC_HUMANOID_WALK_ON_SPOT
+#define IWBC_HUMANOID_WALK_ON_SPOT
+#include <chrono>
+#include <iostream>
+#include <signal.h>
+
+#include <inria_wbc/behaviors/behavior.hpp>
+#include <inria_wbc/controllers/talos_pos_tracker.hpp>
+#include <inria_wbc/estimators/cop.hpp>
+#include <inria_wbc/utils/trajectory_handler.hpp>
+
+namespace inria_wbc {
+    namespace behaviors {
+        namespace humanoid {
+            class Walk : public Behavior {
+            public:
+                Walk(const controller_ptr_t& controller, const YAML::Node& config);
+                Walk() = delete;
+                Walk(const Walk& other) = default;
+                void update(const controllers::SensorData& sensor_data = {}) override;
+
+                std::string behavior_type() const override { return controllers::behavior_types::SINGLE_SUPPORT; };
+
+            private:
+                void _generate_trajectories(int num_of_cycles);
+                int time_ = 0;
+                float dt_;
+                int _current_traj = 0;
+                Eigen::VectorXd _last_com;
+                pinocchio::SE3 _last_lf;
+                pinocchio::SE3 _last_rf;
+
+                std::vector<std::vector<Eigen::VectorXd>> _com_trajs;
+                std::vector<std::vector<pinocchio::SE3>> _lf_trajs;
+                std::vector<std::vector<pinocchio::SE3>> _rf_trajs;
+
+                std::vector<std::vector<pinocchio::SE3>> _lh_trajs;
+                std::vector<std::vector<pinocchio::SE3>> _rh_trajs;
+
+
+                float traj_com_duration_ = 3; //will be changed if specified in yaml
+                float traj_foot_duration_ = 3; //will be changed if specified in yaml
+                float step_height_ = 0.1;
+                float step_length_ = 0.1;
+                int num_of_cycles_ = 10;
+                bool run_ = true;
+                // State machine stats for walking on the spot cycle
+                int state_ = -1;
```

Comment:
better to keep underscore prefix like previous attributes

Reply:
The problem is that the code that comes from Brice ends with _ whereas mine starts with _. RosControl uses underscores at the end, but the rest of the world uses the _ at the beginning (I think it is harder to forget and the _ can also mean "this->"). I do not think we are consistent at the moment...